docs/reference/builder.md

```diff
@@ -109,22 +109,169 @@ Here is the format of the `Dockerfile`:
     # Comment
     INSTRUCTION arguments
 
-The instruction is not case-sensitive, however convention is for them to
-be UPPERCASE in order to distinguish them from arguments more easily.
+The instruction is not case-sensitive. However, convention is for them to
+be UPPERCASE to distinguish them from arguments more easily.
 
-Docker runs the instructions in a `Dockerfile` in order. **The
-first instruction must be \`FROM\`** in order to specify the [*Base
-Image*](glossary.md#base-image) from which you are building.
 
-Docker will treat lines that *begin* with `#` as a
-comment. A `#` marker anywhere else in the line will
-be treated as an argument. This allows statements like:
+Docker runs instructions in a `Dockerfile` in order. **The first 
+instruction must be \`FROM\`** in order to specify the [*Base
+Image*](glossary.md#base-image) from which you are building. 
+
+Docker will treat lines that *begin* with `#` as a comment, unless the line is 
+a valid [parser directive](builder.md#Parser directives). A `#` marker anywhere
+else in a line will be treated as an argument. This allows statements like:
 
     # Comment
     RUN echo 'we are running some # of cool things'
 
-Here is the set of instructions you can use in a `Dockerfile` for building
-images.
+## Parser directives
+
+Parser directives are optional, and affect the way in which subsequent lines 
+in a `Dockerfile` are handled. Parser directives do not add layers to the build,
+and will not be shown as a build step. Parser directives must appear 
+before any instruction in a Dockerfile, and may be included once, and once only,
+in a Dockerfile. Parser directives are written as a special type of comment in
+the form `# DIRECTIVE=value`.  Line continuation characters are not supported in
+parser directives. Hence the following examples are all invalid:
+
+    # Invalid due to line continuation
+    # DIREC \
+    TIVE=value
+or
+
+    # Invalid due to appearing twice
+    # DIRECTIVE=value1
+    FROM ImageName
+    # DIRECTIVE=value2
+    
+or
+
+    # Invalid due to appearing after a builder instruction
+    FROM ImageName
+    # DIRECTIVE=value
+
+The following would be treated as a comment due to the directive not being
+recognised:
+
+    # UNRECOGNISEDDIRECTIVE=value    
+    
+	
+Directives are not case-sensitive.  However, convention is for them to
+be UPPERCASE to distinguish them from other comments more easily. Non 
+line-breaking whitespace is permitted, as is a subsequent comment. Hence 
+these examples are all syntactically valid, and will be treated identically:
+
+    #DIRECTIVE=value
+    # DIRECTIVE =value
+	#	DIRECTIVE= value		# Some ignored comment
```

Comment:
I just noticed this; can we easily make this `# Some ignored comment` invalid? I don't think it's consistent with Dockerfile comments in general because;

> Docker treats lines that begin with # as a comment, unless the line is a valid parser directive. **A `#` marker anywhere else in a line is treated as an argument.**